test/attr/attr_escaping.swift

```diff
@@ -160,3 +160,20 @@ class FooClass {
     set(newValue) { stored = newValue } // expected-error{{cannot assign value of type '(@escaping () -> Int) -> Void' to type 'Optional<(() -> Int) -> Void>'}}
   }
 }
+
+// A call of a closure literal should be non-escaping
+func takesInOut(y: inout Int) {
+  {
+    y += 1 // no-error
+  }()
+}
+
+class HasIVarCaptures {
+  var x: Int = 0
+
+  func method() {
+    {
+      x += 1 // no-error
+    }()
```

Comment:
I believe the bug we were seeing showed up when the closure had an explicit argument list. Can you add a test for that?